tests/ci/worker/init_runner.sh

```diff
@@ -14,14 +14,77 @@ export RUNNER_HOME=/home/ubuntu/actions-runner
 
 export RUNNER_URL="https://github.com/ClickHouse"
 # Funny fact, but metadata service has fixed IP
-INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+INSTANCE_ID=$(ec2metadata --instance-id)
 export INSTANCE_ID
 
 # combine labels
-RUNNER_TYPE=$(/usr/local/bin/aws ec2 describe-tags --filters "Name=resource-id,Values=$INSTANCE_ID" | jq '.Tags[] | select(."Key" == "github:runner-type") | .Value' -r)
+RUNNER_TYPE=$(/usr/local/bin/aws ec2 describe-tags --filters "Name=resource-id,Values=$INSTANCE_ID" --query "Tags[?Key=='github:runner-type'].Value" --output text)
 LABELS="self-hosted,Linux,$(uname -m),$RUNNER_TYPE"
 export LABELS
 
+# Refresh CloudWatch agent config
+aws ssm get-parameter --region us-east-1 --name AmazonCloudWatch-github-runners --query 'Parameter.Value' --output text > /opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json
+systemctl restart amazon-cloudwatch-agent.service
+
+# Refresh teams ssh keys
+TEAM_KEYS_URL=$(aws ssm get-parameter --region us-east-1 --name team-keys-url --query 'Parameter.Value' --output=text)
+curl "${TEAM_KEYS_URL}" > /home/ubuntu/.ssh/authorized_keys2
+chown ubuntu: /home/ubuntu/.ssh -R
+
+
+# Create a pre-run script that will restart docker daemon before the job started
+mkdir -p /tmp/actions-hooks
+cat > /tmp/actions-hooks/pre-run.sh << 'EOF'
+#!/bin/bash
+set -xuo pipefail
+
+echo "Runner's public DNS: $(ec2metadata --public-hostname)"
+EOF
+
+cat > /tmp/actions-hooks/post-run.sh << 'EOF'
+#!/bin/bash
+set -xuo pipefail
+
+terminate-and-exit() {
+  echo "Going to terminate the runner"
+  INSTANCE_ID=$(ec2metadata --instance-id)
+  # We execute it with at to not have it as an orphan process
+  # GH Runners kill all remain processes
+  echo "sleep 10; aws ec2 terminate-instances --instance-ids $INSTANCE_ID" | at now
```

Comment:
Maybe `nohup` will be simpler?

```
nohup "sleep 10; aws ec2 terminate-instances --instance-ids $INSTANCE_ID" >& /dev/null &
```

Reply:
didn't work when I tried it. GH runner just doesn't care and clear everything on exit